contents/blog/why-product-engineering-is-so-fun.md

```diff
@@ -0,0 +1,98 @@
+3. The ability to make _important_ decisions - 5/5
+    - This is simply what PMs are there for. Take the given context, decide what to do with it.
+
+Adding this up we get a score of 12/15. Not bad!
+
+### How fun is Software Engineering? 
+
+1. Challenging problems to solve - 5/5
+    - There is nothing like the dopamine rush you get when you've been working through a problem and you finally get it to work. As a software engineer, this happens usually multiple times per day. 
+2. Support from other highly motivated colleagues - 5/5
+    - Software engineers typically work on a team with at least one other engineer. This team aspect is critical to feeling supported.
+3. The ability to make _important_ decisions - 3/5 (or even 2/5)
+    - While engineers can make important decisions about how something is implemented, they're often completely out of touch with making important decisions about the business. Oftentimes this is because they don't have the context, and they are simply told what to build. 
+
+For Software Engineering we get a score of 12 or 13/15. Also not bad.
+
+### How fun is Product Engineering? 
+
+1. Challenging problems to solve - 5/5
+    - You get the dopamine rush associated with coding, plus the UX problems to solve.
+2. Support from other highly motivated colleagues - 5/5
+    - Similar to software engineering, you often work on a team with other product engineers.
+3. The ability to make _important_ decisions - 5/5
+    - Because you're responsible for working with customers and data to decide what gets built, you're inherently making important decisions, similar to product management.
+
+For Product Engineering we get a score of 15/15. Yes I made this scale up, and yes I've given Product Engineering a perfect score. But you did read the title of this blog post, didn't you? I'm _obviously_ biased.
+
+## So what's the catch?
+
+Ah, there is a catch. There is no perfect score for fun without any downsides. There's a concept of a "fun hangover" in children where they have _such_ a fun day that you know the next day is going to be a complete terror. While slightly different in adults, I'm convinced this concept still applies.
+
+The catch is this:
+
+It's really hard to operate simultaneously in both a product and an engineering mindset. The former is an external mindset, and the latter is an internal mindset. Switching between the two takes effort and it's easy to get stuck in one or the other. You have to be very conscious about which mindset you are in - and which one is most useful to you and the company right now - and make that mindset work for the job that currently needs to be done. Sometimes you're in the wrong mindset, and the progress you need to make becomes slow.
```

Comment:
```suggestion
Ah, there is a catch. There is no perfect score for fun without any downsides. One is the concept of a "fun hangover" in children – where they have _such_ a fun day that you know the next day is going to be a complete terror. While slightly different in adults, I'm convinced this concept still applies.

But here's the big one: 

It's **really hard** to operate in a product **and** an engineering mindset **simultaneously**. 

The former is an external mindset; the latter, an internal one. Switching takes effort, so you need to be conscious about which mindset you're in, and whether it's the right mindset for your active task. Progress becomes slow when you get stuck in the wrong one.
```

1. I've never heard of a "fun hangover" and it makes total sense. 🤯
2. Made some tweaks just to tighten the flow a bit, please check I didn't lose the meaning.
3. I added a semicolon because they rule.